sources/PPU/PPU.cpp

```diff
@@ -424,4 +634,214 @@ namespace ComSquare::PPU
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1NoPriority]);
+			//sprites priority 2
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2Priority]);
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1Priority]);
+			//sprites priority 3
+			break;
+		case 1:
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg3NoPriority]);
+			//sprites priority 0
+			if (!this->_registers._bgmode.mode1Bg3PriorityBit)
+				this->addToMainSubScreen(this->_backgrounds[BgName::bg3Priority]);
+			//sprites priority 1
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2NoPriority]);
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1NoPriority]);
+			//sprites priority 2
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2Priority]);
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1Priority]);
+			//sprites priority 3
+			if (this->_registers._bgmode.mode1Bg3PriorityBit)
+				this->addToMainSubScreen(this->_backgrounds[BgName::bg3Priority]);
+			break;
+		case 2:
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2NoPriority]);
+			//sprites priority 0
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1NoPriority]);
+			//sprites priority 1
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2Priority]);
+			//sprites priority 2
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1Priority]);
+			//sprites priority 3
+			break;
+		case 3:
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2NoPriority]);
+			//sprites priority 0
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1NoPriority]);
+			//sprites priority 1
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2Priority]);
+			//sprites priority 2
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1Priority]);
+			//sprites priority 3
+			break;
+		case 4:
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2NoPriority]);
+			//sprites priority 0
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1NoPriority]);
+			//sprites priority 1
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2Priority]);
+			//sprites priority 2
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1Priority]);
+			//sprites priority 3
+			break;
+		case 5:
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2NoPriority]);
+			//sprites priority 0
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1NoPriority]);
+			//sprites priority 1
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg2Priority]);
+			//sprites priority 2
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1Priority]);
+			//sprites priority 3
+			break;
+		case 6:
+			//sprites priority 0
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1NoPriority]);
+			//sprites priority 1
+			//sprites priority 2
+			this->addToMainSubScreen(this->_backgrounds[BgName::bg1Priority]);
+			//sprites priority 3
+            break;
+		case 7:
+			// Not implemented
```

Comment:
case 7 and default should throw an exception to inform that this is not implemented.